packages/core/src/modules/credentials/protocol/BaseCredentialProtocol.ts

```diff
@@ -138,13 +138,10 @@ export abstract class BaseCredentialProtocol<CFs extends CredentialFormatService
   ): Promise<CredentialExchangeRecord> {
     const { message: credentialProblemReportMessage, agentContext } = messageContext
 
-    const connection = messageContext.assertReadyConnection()
-
     agentContext.config.logger.debug(`Processing problem report with message id ${credentialProblemReportMessage.id}`)
 
     const credentialRecord = await this.getByProperties(agentContext, {
       threadId: credentialProblemReportMessage.threadId,
-      connectionId: connection.id,
     })
```

Comment:
This is dangerous as it allows anyone to send a response to the credential exchange by knowing the thread id. For this reason we don't support problem reports currently for connectionsless exchanges.

If we want to support this, we need to handle it the same as is done in the processRequest message. Which consists of:
- query the record without connection id
- then if the record has a connection id:
  - check if it matches the connection id from the exchange
- if the record does not have a connection id:
  - check that the problem report is authorized based on the out of band exchange (the parentThreadId also need to match)
  - if the incoming message has a connection associated, update the connectionId in the exchange record

This is already implemented like this for processRequest, and I think you can copy most of the logic